src/components/Button/Button.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {useMemo} from 'react';
+import {
+  ActivityIndicator,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+} from 'react-native';
+import {ButtonProps} from './Button.types';
+import {horizontalScale, verticalScale} from '../../utils';
+import {useThemeColor} from '../../hooks';
+import {CustomIcon, CustomIconNames} from '../CustomIcon';
+
+export const Button = (props: ButtonProps) => {
+  const {colors} = useThemeColor();
+
+  const {
+    children,
+    mode,
+    isLoading,
+    style,
+    textStyle,
+    disabled,
+    onPress,
+    icon,
+    iconRight,
+    iconStyle,
+    loaderColor = colors.loader,
+    ...restProps
+  } = props;
+
+  const getStyle = useMemo(() => {
+    if (mode === 'outlined') {
+      return [
+        styles.outlined,
+        {
+          borderColor: colors.primary,
+          backgroundColor: colors.transparent,
+        },
+      ];
+    } else if (mode === 'text') {
+      return [
+        styles.textType,
+        {
+          backgroundColor: colors.transparent,
+        },
+      ];
+    }
+    return null;
+  }, [colors.primary, colors.transparent, mode]);
+
+  const getDisabledBtnStyle = useMemo(() => {
+    if (!disabled || mode === 'text') {
+      return null;
+    } else if (mode === 'default') {
+      return {backgroundColor: colors.disabled};
+    } else if (mode === 'outlined') {
+      return {borderColor: colors.disabled};
+    }
+  }, [colors.disabled, disabled, mode]);
+
+  const disabledLabelStyle = useMemo(() => {
+    if (disabled && mode !== 'default') {
+      return {color: colors.disabled};
+    }
+    return null;
+  }, [colors.disabled, disabled, mode]);
+
+  const getTextStyle = useMemo(() => {
+    if (mode === 'default') {
+      return {color: colors.text2};
+    } else {
+      return {color: colors.text};
+    }
+  }, [colors.text, colors.text2, mode]);
+
+  const renderIcon = (name: CustomIconNames) => {
+    return (
+      <CustomIcon
+        name={name}
+        style={[styles.icon, getTextStyle, disabledLabelStyle, iconStyle]}
+      />
+    );
+  };
+
+  const renderText = () => {
+    return (
+      <Text
+        style={[styles.text, getTextStyle, disabledLabelStyle, textStyle]}
+        lineBreakMode="tail"
+        numberOfLines={1}>
+        {children}
+      </Text>
+    );
+  };
+
+  const ButtonStyles = [
+    [
+      styles.buttonContainer,
+      {
+        backgroundColor: colors.primary,
+      },
+    ],
+    getStyle,
+    getDisabledBtnStyle,
+    iconRight ? styles.rightIcon : null,
+    style,
+  ];
+
+  return (
+    <TouchableOpacity
+      style={ButtonStyles}
+      disabled={disabled || isLoading}
+      onPress={onPress}
+      {...restProps}>
+      {isLoading ? (
+        <ActivityIndicator size="small" color={loaderColor} />
```

Comment:
While showing loader, the button width should not decrease

Reply:
Fixed, I have added loader before the text and icon by seeing at the RN Paper button design when loading.